src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs

```diff
@@ -47,10 +48,12 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// See the README.md in this folder for more information
     /// </remarks>
     [Export(typeof(ITaggerProvider))]
+    [Export(typeof(IDocumentEvents))]
```

Comment:
There are a number of VS APIs that look like they should tell us when a file has been closed.
However, using them would involve a lot more work, particularly to make them follow the VS threading rules for MEF components.

The `TaggerProvider` already tracks all of the documents that we can analyse, and knows when a document has been closed. The simplest implementation that works is the moment is just to raise the event from this class for now.

I've manually tested this by modifying the ConnectedMode package to import `IDocumentEvents` and log when documents are closed, and it works as expected (if there are two or more instances of a window for the same file, the event is not raised until the last window is closed).

See the notes in #4509 for more information.